### PR TITLE
Import tests from Google V8 (Arrow Functions)

### PR DESCRIPTION
**Pull request by rwaldron**
These tests are derived from the following files within the Google V8
project:

	test/mjsunit/harmony/arrow-functions.js

```
test262 --command "v8 --harmony-arrow-functions" Arrow-Function;
test262 --command "js" Arrow-Function;
```

### CODE REVIEW

File: test/language/arrow-function/Arrow-Function_semantics.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.2
+description: >
+    Semantics associated with specific syntactic forms.
```
Comment (domenic) [inline test/language/arrow-function/Arrow-Function_semantics.js]: specific *arrow function* syntactic forms :)

File: test/language/arrow-function/Arrow-Function_rules-for-prototype.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.2
+description: >
+    Arrow functions are like functions, except they throw when using the
```
Comment (domenic) [inline test/language/arrow-function/Arrow-Function_rules-for-prototype.js]: This test seems a bit broader than this sentence implies. Not sure what a good description would be though...

File: test/language/arrow-function/Arrow-Function_semantics.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+var square = x => x * x;
+assert.sameValue(square(3), 9);
+
+// Parenthesize the body to return an object literal expression
+var key_maker = val => ({key: val});
+assert.sameValue(key_maker(empty).key, empty);
+
+// Expression Body implicit return
+var evens = [0, 2, 4, 6, 8];
+assert(compareArray(evens.map(v => v + 1), [1, 3, 5, 7, 9]));
```
Comment (domenic) [inline test/language/arrow-function/Arrow-Function_semantics.js]: We should add `assert.arrayLikeEqual` maybe to encompass this case, before this pattern gets too widespread.

Comment (rwaldron) [inline test/language/arrow-function/Arrow-Function_semantics.js]: I will do that after this and the two array-specific batches land. That way I can do a single pass to update them all.

**Comment by domenic** [comment]
Overall LGTM, nice smoke tests. One could make a case for splitting up into more files, but probably not worth it IMO.

File: test/language/arrow-function/Arrow-Function_semantics.js (hunk 3 above)
Comment (domenic) [inline test/language/arrow-function/Arrow-Function_semantics.js]: :+1:

**Comment by rwaldron** [comment]
>  One could make a case for splitting up into more files,

In cases like this, where there are relatively few tests, I'd like to hold off on big splits until phase 2 when we're actively writing new tests.

**Comment by bterlson** [comment]
Well I'd prefer it to be split but it's probably fine for now.

**Pull request merged**